packages/@lwc/features/src/runtime.ts

```diff
@@ -4,29 +4,42 @@
  * SPDX-License-Identifier: MIT
  * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/MIT
  */
-import { FeatureFlag } from './flags';
-import { assert, create, isFalse, isTrue } from '@lwc/shared';
+import features, { FeatureFlagLookup, FeatureFlagValue } from './flags';
+import { create, isFalse, isTrue, isUndefined } from '@lwc/shared';
 
-interface FeatureFlagConfig {
-    [name: string]: FeatureFlag;
-}
-
-const runtimeFlags: FeatureFlagConfig = create(null);
+const runtimeFlags: FeatureFlagLookup = create(null);
 
-function setFeatureFlag(name: string, value: FeatureFlag) {
-    assert.invariant(
-        isTrue(value) || isFalse(value),
-        `Runtime feature flags must be set to a boolean value but received a(n) ${typeof value} value for the '${name}' flag.`
-    );
+// This function is not whitelisted for use within components and is meant for
+// configuring runtime feature flags during app initialization.
+function setFeatureFlag(name: string, value: FeatureFlagValue) {
+    const isBoolean = isTrue(value) || isFalse(value);
+    if (!isBoolean) {
+        const message = `Invalid ${typeof value} value specified for the "${name}" flag. Runtime feature flags can only be set to a boolean value.`;
+        if (process.env.NODE_ENV === 'production') {
+            console.error(message);
+        } else {
+            throw new TypeError(message);
+        }
+    }
+    if (isUndefined(features[name])) {
+        console.warn(
+            `LWC feature flag "${name}" is undefined. Possible reasons are that 1) it was misspelled or 2) it was removed from the @lwc/features package.`
+        );
+    }
     runtimeFlags[name] = value;
```

Comment:
is there a specific reason to set the flag when `name` is `undefined` (is not a valid flag)?

Reply:
No, it doesn't make a different if it's set or not because our code will never reference it.